temba/channels/handlers.py

```diff
@@ -182,6 +185,41 @@ def post(self, request, *args, **kwargs):
 
         return HttpResponse("Not Handled, unknown action", status=400)  # pragma: no cover
 
+    @classmethod
+    def get_ringing_channel(cls, to_number):
+        return Channel.objects.filter(address=to_number, channel_type=Channel.TYPE_TWIML, role__contains='A', is_active=True).exclude(org=None).first()
+
+    @classmethod
+    def get_received_channel(cls, channel_uuid=None, to_number=None):
+        return Channel.objects.filter(uuid=channel_uuid, is_active=True, channel_type=Channel.TYPE_TWIML).exclude(org=None).first()
+
+    @classmethod
+    def get_client(cls, org):
+        return org.get_twiml_client()
+
+    @classmethod
+    def is_twiml_handler(cls):
+        return True
+
+
+class TwilioHandler(TwimlAPIHandler):
+
+    @disable_middleware
+    def dispatch(self, *args, **kwargs):
+        return super(TwilioHandler, self).dispatch(*args, **kwargs)
+
+    def get_ringing_channel(cls, to_number):
+        return Channel.objects.filter(address=to_number, channel_type=Channel.TYPE_TWILIO, role__contains='A', is_active=True).exclude(org=None).first()
+
+    def get_received_channel(cls, channel_uuid=None, to_number=None):
+        return Channel.objects.filter(address=to_number, is_active=True).exclude(org=None).first()
+
+    def get_client(cls, org):
+        return org.get_twilio_client()
+
+    def is_twiml_handler(cls):
+        return False
```

Comment:
Confusing, Twilio certainly handles twiml. Please rework this abstraction for better clarity. Perhaps a method that validates the connection that is a no-op on TwimlAPIHandler? Though not sure where else this is used.